### PR TITLE
[doc][ml] add one missing train public API reference + lint checker

### PR DESCRIPTION
**Pull request by can-anyscale**
- Add one missing train public API reference. The full list of missing references is here: https://docs.google.com/spreadsheets/d/1wQALnsUuFN4dMgmGITfiVX41fCc6P6FEjYVO6jDWGtM/edit?gid=406692730#gid=406692730
- Enable api policy lint checker as well

Test:
- CI

Screenshot 2024-08-14 at 11 15 52 AM

### CODE REVIEW

File: doc/source/train/api/api.rst [hunk 1]
```diff
@@ -163,6 +163,16 @@ Ray Train Output
 
     ~train.Result
 
+Ray Train Error
```
Comment (matthewdeng) [inline doc/source/train/api/api.rst:166]:
```suggestion
Ray Train Errors
```

File: doc/source/train/api/api.rst [hunk 2]
```diff
@@ -163,6 +163,16 @@ Ray Train Output
...
+---------------
+
+.. autosummary::
+    :nosignatures:
+    :template: autosummary/class_without_autosummary.rst
+    :toctree: doc/
+
+    ~train.error.SessionMisuseError
```
Comment (matthewdeng) [inline doc/source/train/api/api.rst:174]: Let's move `~train.base_trainer.TrainingFailedError` from the section below into this one as well?

File: doc/source/train/api/api.rst
Comment (matthewdeng) [inline doc/source/train/api/api.rst:1]: @justinvyu PTAL!

**Review by aslonnie — APPROVED** [review]

**Review by justinvyu — APPROVED** [review]
Thanks!

**Pull request merged**